runtime/src/main/java/org/corfudb/runtime/view/StreamsView.java

```diff
@@ -129,10 +125,17 @@ public long append(@Nonnull Object object, @Nullable TxResolutionInfo conflictIn
 
             // Is our token a valid type?
             if (tokenResponse.getRespType() == TokenType.TX_ABORT_CONFLICT) {
+                UUID conflictStream = null;
+                byte[] conflictKey = tokenResponse.getConflictKey();
+                if (conflictKey != null && conflictKey.length > 0) {
+                    conflictStream = conflictInfo.getStreamIDForConflictKey(conflictKey);
```

Comment:
What if the same key exists in two streams, it can match the wrong stream right?

Reply:
Yes, you're right. I was trying to avoid sending more info on the TokenResponse, I guess there is no way to go around this, I'll just make the change on the sequencer server side.